From 7089bed6c2c3baa5ba567ddb02d5df451893cf89 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Tue, 8 May 2018 23:38:23 +0200
Subject: [PATCH] ipq40xx: Add SDHCI node to qcom-ipq4019.dtsi

Add SDHCI node that is needed to support SDHCI in IPQ4019 as the device supports is.
---
 arch/arm/boot/dts/qcom-ipq4019.dtsi | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/qcom-ipq4019.dtsi b/arch/arm/boot/dts/qcom-ipq4019.dtsi
index 5c1b097..b90c472 100644
--- a/arch/arm/boot/dts/qcom-ipq4019.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq4019.dtsi
@@ -439,6 +439,33 @@
 			status = "disabled";
 		};
 
+		vccq_sd0: regulator@1948000 {
+			compatible = "qcom,regulator-ipq40xx";
+			regulator-name = "SD0 VccQ";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <3000000>;
+			states = <3000000 0x3
+				  1800000 0x1>;
+			reg = <0x01948000 0x4>;
+			mask = <0x3>;
+			status = "disabled";
+		};
+
+		sdhc_1: sdhci@7824000 {
+			compatible = "qcom,sdhci-msm-v4";
+			reg = <0x7824900 0x11c>, <0x7824000 0x800>;
+			reg-names = "hc_mem", "core_mem";
+			interrupts = <0 123 0>, <0 138 0>;
+			interrupt-names = "hc_irq", "pwr_irq";
+			qcom,bus-width = <8>;
+			qcom,max_clk = <192000000>;
+			clocks = <&gcc GCC_SDCC1_APPS_CLK>,
+				 <&gcc GCC_SDCC1_AHB_CLK>;
+			clock-names = "core", "iface";
+			qcom,large-address-bus;
+			status = "disabled";
+		};
+
 		qpic_bam: dma@7984000 {
 			compatible = "qcom,bam-v1.7.0";
 			reg = <0x7984000 0x1a000>;
